src/main/java/picard/sam/DuplicationMetrics.java

```diff
@@ -24,62 +24,84 @@
 import htsjdk.samtools.util.Histogram;
+import picard.analysis.replicates.MergeableMetricBase;
 
 /**
  * Metrics that are calculated during the process of marking duplicates
  * within a stream of SAMRecords.
  */
-public class DuplicationMetrics extends MetricBase {
+public class DuplicationMetrics extends MergeableMetricBase {
     /** The library on which the duplicate marking was performed. */
+    @MergeByAssertEquals
     public String LIBRARY;
 
     /**
      * The number of mapped reads examined which did not have a mapped mate pair,
      * either because the read is unpaired, or the read is paired to an unmapped mate.
      */
+    @MergeByAdding
     public long UNPAIRED_READS_EXAMINED;
 
     /** The number of mapped read pairs examined. (Primary, non-supplemental) */
+    @MergeByAdding
     public long READ_PAIRS_EXAMINED;
 
     /** The number of reads that were either secondary or supplementary */
+    @MergeByAdding
     public long SECONDARY_OR_SUPPLEMENTARY_RDS;
 
     /** The total number of unmapped reads examined. (Primary, non-supplemental) */
+    @MergeByAdding
     public long UNMAPPED_READS;
 
     /** The number of fragments that were marked as duplicates. */
+    @MergeByAdding
     public long UNPAIRED_READ_DUPLICATES;
 
     /** The number of read pairs that were marked as duplicates. */
+    @MergeByAdding
     public long READ_PAIR_DUPLICATES;
 
     /**
      * The number of read pairs duplicates that were caused by optical duplication.
      * Value is always < READ_PAIR_DUPLICATES, which counts all duplicates regardless of source.
      */
+    @MergeByAdding
     public long READ_PAIR_OPTICAL_DUPLICATES;
 
     /** The fraction of mapped sequence that is marked as duplicate. */
+    @NoMergingIsDerived
     public Double PERCENT_DUPLICATION;
 
     /** The estimated number of unique molecules in the library based on PE duplication. */
+    @NoMergingIsDerived
     public Long ESTIMATED_LIBRARY_SIZE;
 
     /**
      * Fills in the ESTIMATED_LIBRARY_SIZE based on the paired read data examined where
      * possible and the PERCENT_DUPLICATION.
      */
-    public void calculateDerivedMetrics() {
+    @Override
+    public void calculateDerivedFields() {
         this.ESTIMATED_LIBRARY_SIZE = estimateLibrarySize(this.READ_PAIRS_EXAMINED - this.READ_PAIR_OPTICAL_DUPLICATES,
-                                                          this.READ_PAIRS_EXAMINED - this.READ_PAIR_DUPLICATES);
+                this.READ_PAIRS_EXAMINED - this.READ_PAIR_DUPLICATES);
 
         PERCENT_DUPLICATION = (UNPAIRED_READ_DUPLICATES + READ_PAIR_DUPLICATES *2) /(double) (UNPAIRED_READS_EXAMINED + READ_PAIRS_EXAMINED *2);
     }
 
     /**
+     * Fills in the ESTIMATED_LIBRARY_SIZE based on the paired read data examined where
+     * possible and the PERCENT_DUPLICATION.
+     *
+     * Deprecated, use {@link #calculateDerivedFields()} instead.
+     */
+    @Deprecated
+    public void calculateDerivedMetrics() {
```

Comment:
Can you change the few other places (in picard) where the newly deprecated function is called?